wolfcrypt/src/asn.c

```diff
@@ -3883,9 +3609,27 @@ int wc_EncryptPKCS8Key(byte* key, word32 keySz, byte* out, word32 outSz,
     }
     if (ret == 0) {
         /* outer = SEQ [ pbe ] OCT encrypted_PKCS#8_key */
+        /* plus 2 for tag and length for pbe */
         outerLen = 2 + pbeLen;
-        outerLen += SetOctetString(keySz + padSz, out);
         outerLen += keySz + padSz;
+        if (out == NULL) {
+            /* plus 2 tags (for sequence and octet string). */
+            *outSz = outerLen + 2;
+            /* plus bytes needed for encoding octet string length. */
+            tmp = SetLength(keySz + padSz, NULL);
+            *outSz += tmp;
+            /* plus bytes needed for encoding sequence length:
+                 1 -> octet string tag
+                 tmp -> length of octet string length encoding
+                 outerLen -> everything else */
+            *outSz += SetLength(outerLen + 1 + tmp, NULL);
+        #ifdef WOLFSSL_SMALL_STACK
+            if (saltTmp != NULL)
+                XFREE(saltTmp, heap, DYNAMIC_TYPE_TMP_BUFFER);
```

Comment:
Do we need saltTmp allocated yet? If not please move the allocation to after the length only.